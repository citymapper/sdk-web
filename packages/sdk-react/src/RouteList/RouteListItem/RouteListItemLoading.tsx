import React from 'react'
import LoadingSkeleton from '../../LoadingSkeleton'
import { Row } from './RouteListItemDetails'

interface RouteListItemLoadingProps {
  index?: number
}

const RouteListItemLoading: React.FC<RouteListItemLoadingProps> = ({
  index = 0,
}) => {
  const animationDelay = index * 100
  return (
    <>
      <Row>
        <LoadingSkeleton
          width={100}
          height={24}
          animationDelay={animationDelay}
        />
        <LoadingSkeleton
          width={50}
          height={20}
          animationDelay={animationDelay}
        />
      </Row>
      <Row>
        <LoadingSkeleton
          width="75%"
          height={12}
          animationDelay={animationDelay}
        />
        <LoadingSkeleton
          width="15%"
          height={12}
          animationDelay={animationDelay}
        />
      </Row>
    </>
  )
}

export default RouteListItemLoading
