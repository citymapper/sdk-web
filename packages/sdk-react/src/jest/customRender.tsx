import { render, RenderOptions } from '@testing-library/react'
import React, { JSXElementConstructor, ReactElement } from 'react'
import CMProvider from '../CMProvider'

const TestWrapper: React.FC<{ children: React.ReactElement }> = ({
  children,
}) => (
  <CMProvider
    config={{
      apiKey: '',
      citymapperFont: false,
    }}
  >
    {children}
  </CMProvider>
)

const customRender = (
  ui: ReactElement<any, string | JSXElementConstructor<any>>,
  options?: RenderOptions<
    typeof import('@testing-library/dom/types/queries'),
    HTMLElement,
    HTMLElement
  >
) => render(ui, { wrapper: TestWrapper, ...options })

export default customRender
