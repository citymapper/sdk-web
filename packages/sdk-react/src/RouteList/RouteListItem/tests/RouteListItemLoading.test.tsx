import React from 'react'
import customRender from '../../../jest/customRender'
import RouteListItemLoading from '../RouteListItemLoading'

describe('RouteListItemLoading', () => {
  it('should match snapshot without delays', () => {
    const { container } = customRender(<RouteListItemLoading index={0} />)
    expect(container).toMatchSnapshot()
  })

  it('should match snapshot with delay', () => {
    const { container } = customRender(<RouteListItemLoading index={5} />)
    expect(container).toMatchSnapshot()
  })
})
