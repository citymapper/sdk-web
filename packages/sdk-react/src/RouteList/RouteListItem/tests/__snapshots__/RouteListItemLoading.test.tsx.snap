// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteListItemLoading should match snapshot with delay 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  display: inline-block;
  height: 24px;
  width: 100px;
  position: relative;
  overflow: hidden;
  background-color: #dfe3e9;
  -webkit-animation: shimmer 3s infinite;
  animation: shimmer 3s infinite;
  -webkit-animation-delay: 500ms;
  animation-delay: 500ms;
}

.emotion-2 {
  display: inline-block;
  height: 20px;
  width: 50px;
  position: relative;
  overflow: hidden;
  background-color: #dfe3e9;
  -webkit-animation: shimmer 3s infinite;
  animation: shimmer 3s infinite;
  -webkit-animation-delay: 500ms;
  animation-delay: 500ms;
}

.emotion-4 {
  display: inline-block;
  height: 12px;
  width: 75%;
  position: relative;
  overflow: hidden;
  background-color: #dfe3e9;
  -webkit-animation: shimmer 3s infinite;
  animation: shimmer 3s infinite;
  -webkit-animation-delay: 500ms;
  animation-delay: 500ms;
}

.emotion-5 {
  display: inline-block;
  height: 12px;
  width: 15%;
  position: relative;
  overflow: hidden;
  background-color: #dfe3e9;
  -webkit-animation: shimmer 3s infinite;
  animation: shimmer 3s infinite;
  -webkit-animation-delay: 500ms;
  animation-delay: 500ms;
}

<div>
  <div
    class="cm-route-list-item-row emotion-0"
  >
    <div
      class="cm-loading-skeleton emotion-1"
      height="24"
      width="100"
    />
    <div
      class="cm-loading-skeleton emotion-2"
      height="20"
      width="50"
    />
  </div>
  <div
    class="cm-route-list-item-row emotion-0"
  >
    <div
      class="cm-loading-skeleton emotion-4"
      height="12"
      width="75%"
    />
    <div
      class="cm-loading-skeleton emotion-5"
      height="12"
      width="15%"
    />
  </div>
</div>
`;

exports[`RouteListItemLoading should match snapshot without delays 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  display: inline-block;
  height: 24px;
  width: 100px;
  position: relative;
  overflow: hidden;
  background-color: #dfe3e9;
  -webkit-animation: shimmer 3s infinite;
  animation: shimmer 3s infinite;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
}

.emotion-2 {
  display: inline-block;
  height: 20px;
  width: 50px;
  position: relative;
  overflow: hidden;
  background-color: #dfe3e9;
  -webkit-animation: shimmer 3s infinite;
  animation: shimmer 3s infinite;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
}

.emotion-4 {
  display: inline-block;
  height: 12px;
  width: 75%;
  position: relative;
  overflow: hidden;
  background-color: #dfe3e9;
  -webkit-animation: shimmer 3s infinite;
  animation: shimmer 3s infinite;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
}

.emotion-5 {
  display: inline-block;
  height: 12px;
  width: 15%;
  position: relative;
  overflow: hidden;
  background-color: #dfe3e9;
  -webkit-animation: shimmer 3s infinite;
  animation: shimmer 3s infinite;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
}

<div>
  <div
    class="cm-route-list-item-row emotion-0"
  >
    <div
      class="cm-loading-skeleton emotion-1"
      height="24"
      width="100"
    />
    <div
      class="cm-loading-skeleton emotion-2"
      height="20"
      width="50"
    />
  </div>
  <div
    class="cm-route-list-item-row emotion-0"
  >
    <div
      class="cm-loading-skeleton emotion-4"
      height="12"
      width="75%"
    />
    <div
      class="cm-loading-skeleton emotion-5"
      height="12"
      width="15%"
    />
  </div>
</div>
`;
